io/pci/PowerVMEEH.py

```diff
@@ -83,21 +87,46 @@ def test_eeh_basic_pe(self):
             self.log.info("PCI mask = %s" % self.pci_mask)
             self.log.info("PCI class name = %s" % self.pci_class_name)
             self.log.info("PCI interface = %s" % self.pci_interface)
-            for func in self.function:
-                self.log.info("Running error injection on pe %s function %s"
-                              % (self.addr, func))
-                self.return_code = self.basic_eeh(func, self.pci_class_name,
-                                                  self.pci_interface,
-                                                  self.pci_mem_addr,
-                                                  self.pci_mask)
-                if self.return_code == EEH_MISS:
-                    continue
-                else:
-                    if not self.check_eeh_pe_recovery(self.addr):
-                        self.fail("PE %s recovery failed after first EEH"
-                                  % self.addr)
+            while self.num_of_hit <= self.max_freeze:
+                for func in self.function:
+                    self.log.info("Running error inject on pe %s function %s"
+                                  % (self.addr, func))
+                    if self.num_of_miss < 5:
```

Comment:
5 is just a random number. I have seen many times where eeh_hit is missed for first three times, and then is successful after that, and also once its not hit even after 5 times, its safe to assume to try with next function. Even in the attached job.log file, EEH is miss for first three times, but then later its successful for first adapter.